src/bots/fillerBulk.ts

```diff
@@ -0,0 +1,146 @@
+import {
+	NodeToFill,
+	NodeToTrigger,
+	PerpMarketAccount,
+	DLOB,
+	calculateAskPrice,
+	calculateBidPrice,
+	MarketType,
+	BN,
+	BN_MAX,
+	ZERO,
+} from '@drift-labs/sdk';
+
+import { FillerLiteBot } from './fillerLite';
+
+const MAX_NUM_MAKERS = 6;
+
+export class FillerBulkBot extends FillerLiteBot {
+	protected getPerpNodesForMarket(
+		market: PerpMarketAccount,
+		dlob: DLOB
+	): {
+		nodesToFill: Array<NodeToFill>;
+		nodesToTrigger: Array<NodeToTrigger>;
+	} {
+		const marketIndex = market.marketIndex;
+
+		const oraclePriceData =
+			this.driftClient.getOracleDataForPerpMarket(marketIndex);
+
+		const fillSlot = this.orderSubscriber.getSlot();
```

Comment:
might be good to do this.dlobSource.getSlot()

Reply:
added new getMaxSlot since slot subscriber can fall behind